hcmanager/labels.go

```diff
@@ -0,0 +1,101 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+package hcmanager
+
+import (
+	"context"
+	"errors"
+
+	bmh "github.com/metal3-io/baremetal-operator/pkg/apis/metal3/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+//DeleteHWCCLabel deletes the hwcc label from the baremetal host
+func (mgr HardwareClassificationManager) DeleteHWCCLabel(ctx context.Context, hcMetaData v1.ObjectMeta, hosts bmh.BareMetalHostList) []string {
+	var deleteLabelError []string
+	for _, host := range hosts.Items {
+		if err := mgr.DeleteLabels(ctx, hcMetaData, host); err != nil {
+			deleteLabelError = append(deleteLabelError, host.Name+" "+err.Error())
+		}
+	}
+	return deleteLabelError
+}
+
+// DeleteLabels delete existing label of the baremetal host
+func (mgr HardwareClassificationManager) DeleteLabels(ctx context.Context, hcMetaData v1.ObjectMeta, host bmh.BareMetalHost) error {
```

Comment:
The naming here is a bit inconsistent with next function. Maybe you should add the loop over the bmh in this function too and make it a higher level function, so that you could call it from the Reconcile alongside the SetLabels, rather than from within the SetLabels. This way would be clearer and more readable